### PR TITLE
add unittest framework for neo-devpack-dotnet branch 2.x.

### PR DESCRIPTION
**Pull request by lightszero**
Add unittest framework for master-2.x.
fix bug will be more simple.

### CODE REVIEW

File: Neo.Compiler.MSIL.UnitTests/UnitTest_Appcall.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+using Neo.VM;
+
+namespace Neo.Compiler.MSIL
+{
+    [TestClass]
+    public class UnitTest_AppCall
+    {
+        [TestMethod]
+        public void Test_Appcall()
+        {
```
Comment (shargon) [inline Neo.Compiler.MSIL.UnitTests/UnitTest_Appcall.cs]: Better remove this file

File: Neo.Compiler.MSIL.UnitTests/UnitTest1.cs [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
+            foreach (var c in nefMethod.body_Codes)
             {
                 Console.WriteLine(c.Key.ToString("X04") + "=>" + c.Value.ToString());
             }
         }
 
         private static void DumpBytes(byte[] data)
         {
-            Console.WriteLine("AVM=");
+            Console.WriteLine("NEF=");
```
Comment (lock9) [inline Neo.Compiler.MSIL.UnitTests/UnitTest1.cs]: Hi, I think we should keep AVM name for 2.x

File: Neo.Compiler.MSIL.UnitTests/UnitTest1.cs [hunk 3]
```diff
@@ -41,30 +41,38 @@ public void GetAllILFunction()
         [TestMethod]
         public void TestDumpAFunc()
         {
-            var ilmethod = testtool.FindMethod("TestClass1", "UnitTest_001");
+            var testtool = NeonTestTool.BuildScript("./TestClasses/Contract1.cs");
+            var ilmethod = testtool.FindMethod("Contract1", "UnitTest_001");
             var neomethod = testtool.GetNEOVMMethod(ilmethod);
-            DumpAVM(neomethod);
+            DumpNEF(neomethod);
```
Comment (lock9) [inline Neo.Compiler.MSIL.UnitTests/UnitTest1.cs]: I think it is better to use AVM in this branch

File: Neo.Compiler.MSIL.UnitTests/UnitTest1.cs [hunk 4]
```diff
@@ -1,26 +1,25 @@
...
     public class UnitTest1
     {
-        private static readonly NeonTestTool testtool = new NeonTestTool("net4smartcontract.test.dll");
-
-        private static void DumpAVM(NeoMethod avmMethod)
+        private static void DumpNEF(NeoMethod nefMethod)
         {
-            Console.WriteLine("dump:" + avmMethod.displayName + " addr in avm:" + avmMethod.funcaddr);
-            foreach (var c in avmMethod.body_Codes)
+            Console.WriteLine("dump:" + nefMethod.displayName + " addr in nef:" + nefMethod.funcaddr);
```
Comment (lock9) [inline Neo.Compiler.MSIL.UnitTests/UnitTest1.cs]: I think we should use avm in this branch

File: Neo.Compiler.MSIL.UnitTests/UnitTest1.cs (hunk 3 above)
Comment (shargon) [reply Neo.Compiler.MSIL.UnitTests/UnitTest1.cs]: Yes, neo2 use avm

**Comment by shargon** [comment]
I can't merge it with this travis bug...

**Comment by lapetina9** [comment]
## Neo PR Test Report

**Repository:** neo-devpack-dotnet 
**Number:** 121 
**Title:** Neo 2 devpack tests     
**Change description:** New unit tests added, including new testing classes.
## Code version / branch

**Neo:**  master  
**Neo Devpack:** https://github.com/neo-project/neo-devpack-dotnet/pull/121 

## Test Results

| Test Name | Network | OS | Test Result | Comments |
| ------- |:-------:| --:| -----------:| --------:|
| Initial UT | UT | Localhost | OK | 

#### Additional Info
Only new unit tests were included so we evaluated that there was no need to deploy it into any network.

## **Test List**

**1. Initial UT**  
Run solution unit tests.

**Comment by shargon** [comment]
@erikzhang travis is stuck 😔, could you merge it please?

**Comment by shargon** [comment]
Travis doesn't start, i don't know why

**Pull request merged**